addon/mixins/components/svg-handling.js

```diff
@@ -0,0 +1,78 @@
+import Mixin from '@ember/object/mixin';
+import { computed } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import { isNone } from '@ember/utils';
+
+/*
+ * This mixin is intended for use with components that render an SVG icon
+ * via the `svg-jar` helper.
+ *
+ * The Polaris React implementation checks the
+ * `fill` attribute on SVG elements and their children to ensure SVGs
+ * always respect whatever `color` they're passed.
+ *
+ * In this mixin, we automate that process - all that's needed is to
+ * include this mixin in the component that renders the SVG, and pass
+ * `id=svgElementId` to the `svg-jar` invocation.
+ *
+ * TODO: this currently only supports having a single SVG per component.
+ * This should be fine for now though since this mixin only gets used
+ * in components whose sole responsibility is to render a single SVG.
+ */
+export default Mixin.create({
+  /**
+   * Generated unique ID to be set on the rendered SVG element.
+   *
+   * @property svgElementId
+   * @type {string}
+   */
+  svgElementId: computed(function() {
+    // We append '-svg' to the component's Ember-generated GUID,
+    // because Ember will automatically use that GUID as the ID
+    // for the root element of non-tagless components.
+    return `${ guidFor(this) }-svg`;
+  }).readOnly(),
+
+  /**
+   * Volatile CP to find the SVG element for this component.
+   *
+   * @property svgElement
+   * @type {DOMNode}
+   */
+  svgElement: computed(function() {
+    return document.querySelector(`#${ this.get('svgElementId') }`);
+  }).volatile(),
```

Comment:
does this needs to be volatile?

Reply:
I figured that this way, it's basically the same as making `document.querySelector` call inline, just a bit cleaner. This means that we're checking against the DOM each time we try using `svgElement`, rather than caching an element that could get removed from the DOM at some point (e.g. if the icon is wrapped in an `if`-condition etc.). It shouldn't affect anything as things currently are, but this way just seemed a bit safer for the future.

If you don't like the `volatile` modifier, I can just make the `querySelector` call inline instead... this just seemed like a nice way of wrapping up finding the SVG element 😛

Reply:
k, yeah, that makes sense...I didn't thought we need it right now for our use-cases...that's why I asked
I'm fine with leaving it for now...maybe add a comment for future-us 😄